api/metrics/views.py

```diff
@@ -0,0 +1,196 @@
+from datetime import timedelta, datetime
+
+import pytz
+from django.utils import timezone
+from django.http import JsonResponse
+from rest_framework.exceptions import ValidationError
+from rest_framework import permissions as drf_permissions
+from elasticsearch.exceptions import NotFoundError, RequestError
+
+from osf.models import Preprint
+from framework.auth.oauth_scopes import CoreScopes
+from api.base.permissions import TokenHasScope
+from osf.metrics import PreprintDownload, PreprintView
+from api.metrics.permissions import IsPreprintMetricsUser
+from api.metrics.serializers import PreprintMetricSerializer
+from api.base.views import JSONAPIBaseView
+
+
+class PreprintMetricMixin(JSONAPIBaseView):
+    permission_classes = (
+        drf_permissions.IsAuthenticated,
+        drf_permissions.IsAdminUser,
+        IsPreprintMetricsUser,
+        TokenHasScope,
+    )
+
+    required_read_scopes = [CoreScopes.METRICS_READ]
```

Comment:
Very much wasn't sure what to do with required scopes, better solution suggestions very welcome!